src/wasm/wasm-ir-builder.cpp

```diff
@@ -431,6 +431,26 @@ Result<> IRBuilder::visitBreak(Break* curr, std::optional<Index> label) {
   return Ok{};
 }
 
+Result<> IRBuilder::visitBreakWithType(Break* curr, Type type) {
+  if (curr->condition) {
+    auto cond = pop();
+    CHECK_ERR(cond);
+    curr->condition = *cond;
+  }
+  if (!curr->value) {
```

Comment:
The `type` should be the only source of truth here.
```suggestion
  if (type == Type::None) {
```

Reply:
done